src/training/save_states.rs

```diff
@@ -266,10 +266,12 @@ pub unsafe fn save_states(module_accessor: &mut app::BattleObjectModuleAccessor)
                     let article: u64 = ArticleModule::get_article(module_accessor, article_idx);
                     let article_object_id =
                         Article::get_battle_object_id(article as *mut app::Article);
-                    ArticleModule::remove_exist_object_id(
-                        module_accessor,
-                        article_object_id as u32,
-                    );
+                    if fighter_kind != *FIGHTER_KIND_PICKEL || article_idx != *FIGHTER_PICKEL_GENERATE_ARTICLE_TABLE { // don't remove crafting table
```

Comment:
Firstly, shouldn't this be "if is_steve AND is crafting table"? I see `or` right now.

Then, to avoid huge nested blocks and keep main logic as unnested as possible, usually blocklists are done like this within loops: 
```rust
// Don't remove crafting table
if fighter_kind == *FIGHTER_KIND_PICKEL && article_idx == *FIGHTER_PICKEL_GENERATE_ARTICLE_TABLE { 
    continue;
}

ArticleModule::remove_exist_object_id(
    module_accessor,
    article_object_id as u32,
);
```

In this way, it can be refactored into a blocklist later too if we have more items to keep.
```rust
// and  we have a static ITEM_REMOVE_BLOCKLIST = [(FIGHTER_KIND_PICKEL, blah)]
if (fighter_kind, article_idx) in ITEM_REMOVE_BLOCKLIST { 
    continue;
}

ArticleModule::remove_exist_object_id(
    module_accessor,
    article_object_id as u32,
);
```

Lastly, this is actually written in functional style, so we should probably use a function to do this instead.
```rust
(0..=0x25)
    // Don't remove crafting table
    .filter(|article_idx| fighter_kind != *FIGHTER_KIND_PICKEL && article_idx != *FIGHTER_PICKEL_GENERATE_ARTICLE_TABLE)
    .filter(|article_idx| ArticleModule::is_exist(module_accessor, article_idx))
    .map(|article_idx| {
        let article: u64 = ArticleModule::get_article(module_accessor, article_idx);
        let article_object_id =
            Article::get_battle_object_id(article as *mut app::Article);
        article_object_id
    })
    .for_each(|article_object_id|
        ArticleModule::remove_exist_object_id(
            module_accessor,
            article_object_id as u32,
        );
     });
```